### PR TITLE
projects/Amlogic: add support for S912 devices

### PR DESCRIPTION
**Pull request by adamg88**
This PR adds support for devices with S912 soc.

### CODE REVIEW

**Comment by kszaq** [comment]
Thank you for submitting this PR! I will add a few suggestions shortly.

**Comment by kszaq** [comment]
Please drop device-specific `linux.aarch64.conf`. We can simply enable HMP- and DEVFREQ- related options in project config, no need to have a separate config to maintain. Amlogic provides a single config for all S9* devices and co can we.

File: projects/Amlogic/devices/S912/packages/libhybris/package.mk [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+PKG_SECTION="devel"
+PKG_SHORTDESC="libhybris: Allows to run bionic-based HW adaptations in glibc systems - libs"
+PKG_TOOLCHAIN="autotools"
+
+PKG_CONFIGURE_OPTS_TARGET="--enable-arch=$TARGET_ARCH \
+                           --with-default-egl-platform=fbdev \
+                           --with-android-headers=$BUILD/android-headers-25 \
+                           --with-default-hybris-ld-library-path=/system/lib \
+                           --enable-mali-quirks"
+post_makeinstall_target(){
```
Comment (kszaq) [inline projects/Amlogic/devices/S912/packages/libhybris/package.mk]: Please drop `post_makeinstall_target()` in favour of updated fbdev patch that you can find here:
https://raw.githubusercontent.com/kszaq/LibreELEC.tv/8.2.2.3/packages/devel/libhybris/patches/libhybris-0003-add-fbdev-window-h.patch

Comment (adamg88) [reply projects/Amlogic/devices/S912/packages/libhybris/package.mk]: I thought the idea going forward within LE was to reduce the number of patches?

**Comment by adamg88** [comment]
As for the changes to `linux.aarch64.conf` I tried what you suggested after your last PR and I encountered some issues, one being that only 4 CPU's was being registered instead of 8 which was probably down to
```-CONFIG_NR_CPUS=8
+CONFIG_NR_CPUS=4
```

File: projects/Amlogic/devices/S912/packages/libhybris/package.mk (hunk 1 above)
Comment (kszaq) [reply projects/Amlogic/devices/S912/packages/libhybris/package.mk]: The patch I linked is already included in your PR but in a different and overblown version. I suggest replacing it with a better version that would also allow you to drop `post_makeinstall_target`.

Comment (adamg88) [reply projects/Amlogic/devices/S912/packages/libhybris/package.mk]: I've just noticed that thanks :) just waiting for compile to finish before updating PR.

**Comment by adamg88** [comment]
PR updated, thanks for recommendations @kszaq

**Comment by kszaq** [comment]
Thank you for updating the PR. 

There are still a few kernel options missing, e.g. `CONFIG_FB_OSD_SUPPORT_SYNC_FENCE `, `CONFIG_PSTORE`, `CONFIG_ASHMEM `, `CONFIG_SW_SYNC`, perhaps something else. 

You don't use MM kernel in your community builds, do you? What I would like to suggest is that I will update projects/Amlogic kernel config to work with S912 devices as soon as we switch to updated kernel in LE master (to this: https://kszaq.libreelec.tv/temp/linux.aarch64.conf), then we can merge this PR without having to update kernel config.

**Comment by adamg88** [comment]
The config at present is for Nougat which I copied over by accident during the refactoring from S905>Amlogic which we are not using in LE/master, the proposed to changes are for compatibility with the kernel in use at present just to give it a booting image and are no way respective of what options the final config may contain, however saying that the above quotes options are not something I have used in any config and have had no issues.

I have no issues with you wanting to submit a further PR to update or maintain the config in the future, it's there for anybody to contribute to.

I just want to get the userspace side of things finished off so work can then be focused on the kernel and I think we are almost there with it after this PR.

**Comment by adamg88** [comment]
@kszaq your kernel config breaks the build and so does your libhybris patch, I have reverted the changes you recommended and it builds fine

Log for your reference, https://pastebin.com/raw/Liz8eX2Y

**Comment by adamg88** [comment]
PR is complete, if there is no further reviews.

**Comment by adamg88** [comment]
Thanks for your changes @kszaq

They have cleaned the PR considerably, do you want me to squash your changes or leave as is

**Comment by kszaq** [comment]
@adamg88 Why have you closed this PR?

**Comment by adamg88** [comment]
Closed for you to take it over, I was told your looking for a way to remove the hacks and set windows resolution properly with libhybris

**Pull request closed without merging**